src/App.tsx

```diff
@@ -1,155 +1,88 @@
-              />
-            </form>
-
-            <div className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<TodoType[]>([]);
+  const [selectedSortType, setSortType] = useState(SortType.All);
+  const [isError, setError] = useState(false);
 
-          {/* This todo is in loadind state */}
-          <div className="todo">
-            <label className="todo__status-label">
-              <input type="checkbox" className="todo__status" />
-            </label>
+  const fetchTodos = async () => {
+    try {
+      const actualTodos = await getTodos(USER_ID);
 
-            <span className="todo__title">Todo is being saved now</span>
-            <button type="button" className="todo__remove">×</button>
+      setTodos(actualTodos);
+    } catch (error) {
+      setError(true);
 
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-        </section>
+      setTimeout(() => {
+        setError(false);
+      }, 3000);
+    }
+  };
 
-        {/* Hide the footer if there are no todos */}
-        <footer className="todoapp__footer">
-          <span className="todo-count">
-            3 items left
-          </span>
+  useEffect(() => {
+    fetchTodos();
+  }, []);
 
-          {/* Active filter should have a 'selected' class */}
-          <nav className="filter">
-            <a href="#/" className="filter__link selected">
-              All
-            </a>
+  const getSortType = (newSortType: SortType) => {
+    setSortType(newSortType);
+  };
 
-            <a href="#/active" className="filter__link">
-              Active
-            </a>
+  const visibleList = useMemo(() => getReorderedList(selectedSortType, todos),
+    [selectedSortType, todos]);
 
-            <a href="#/completed" className="filter__link">
-              Completed
-            </a>
-          </nav>
+  if (!USER_ID) {
+    return <UserWarning />;
+  }
 
-          {/* don't show this button if there are no completed todos */}
-          <button type="button" className="todoapp__clear-completed">
-            Clear completed
-          </button>
-        </footer>
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+        <Header
+          activeTodos={getReorderedList(SortType.Active, todos).length}
+          hasTodos={!!todos.length}
+        />
+        <Section
+          todos={visibleList}
+        />
+        {!!todos.length && (
+          <Footer
+            itemsLeft={getReorderedList(SortType.Active, todos)}
```

Comment:
You call the getReorderedList function several times. Just move the result to a variable and use it